examples/tide/src/main.rs

```diff
@@ -1,24 +1,106 @@
-// And finally, include the generated code for templates and static files.
-include!(concat!(env!("OUT_DIR"), "/templates.rs"));
-
+//! An example of how ructe can be used with the tide framework.
 mod ructe_tide;
 use ructe_tide::Render;
 
-use tide::{Response, StatusCode};
+use httpdate::fmt_http_date;
+use std::future::Future;
+use std::io::{self, Write};
+use std::pin::Pin;
+use std::str::FromStr;
+use std::time::{Duration, SystemTime};
+use templates::statics::{cloud_svg, StaticFile};
+use tide::http::headers::EXPIRES;
+use tide::http::{mime, Error};
+use tide::{Next, Redirect, Request, Response, StatusCode};
 
+/// Main entry point.
+///
+/// Set up an app and start listening for requests.
 #[async_std::main]
 async fn main() -> Result<(), std::io::Error> {
     let mut app = tide::new();
-
-    app.at("/").get(|_| async {
-        let mut res = Response::new(StatusCode::Ok);
-        res.render_html(|o| Ok(templates::hello(o, "world")?))?;
-        Ok(res)
-    });
+    app.middleware(handle_error);
+    app.at("/static/*path").get(static_file);
+    app.at("/favicon.ico")
+        .get(Redirect::new(format!("/static/{}", cloud_svg.name)));
```

Comment:
Redirect for favicon seems a bad pattern would be good if it can send the file directly.

This could encourage people from doing the same for their projects.

Reply:
Hm ...  I usually do redirect for the favicon in real projects.  But yes, when I think about it, it should be very small, so it might be better to serve it directly even if that duplicates the file.

Reply:
Ok, I made a direct handler for the favicon.